### DISCUSSION TITLE
Embed bpmn models through bpmn.js

### DISCUSSION THREAD

**Discussion by stereo9** [outline/outline#3673] [Ideas]
Hi

we are using outline to document our internal processes. 
So far, the only options are using screenshots of a given process or the cawemo integration.
This could be nicely improved by implementing www.bpmn.io/toolkit/bpmn-js/ which would allow for an advanced embed and editing function

**Comment by extead** [comment]
Hello! We also use Outline to store our internal processes. I decided to try implementing diagram rendering based on the integration of mermaidjs by @tommoor. I managed to create basic functionality to test the integration feasibility. Currently, it supports displaying a diagram based on existing diagram code.

You can see the changes here: https://github.com/outline/outline/compare/main...extead:outline:bpmnjs
There's a lot of unnecessary code in the current implementation, and I would appreciate help in completing the implementation since I'm not a professional JS developer.

![ezgif com-video-to-gif](https://github.com/outline/outline/assets/2337077/03da4918-8ed3-4ecf-9f1b-4123c2ebb90b)
